Build failure with GCC (>= ?) 5.4

```
../../src/log.h:41:34: error: ISO C does not support '__FUNCTION__' predefined identifier [-Werror=pedantic]
     log_impl(__LINE__, __FILE__, __FUNCTION__, level, __VA_ARGS__);
```
so perhaps `error=pedantic` should only be for developers (rather than those installing/packaging)? i.e. `TEST_AUTHOR=1`'ish

--- a/src/log.h
+++ b/src/log.h
@@ -1,5 +1,6 @@
 /**
- * Copyright (C) 2013, 2014 Genome Research Ltd. All rights reserved.
+ * Copyright (C) 2013, 2014, 2017 Genome Research Ltd. All rights
+ * reserved.
  *
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -38,7 +39,7 @@ typedef enum {
 } log_level;
 
 #define logmsg(level, ...) \
-    log_impl(__LINE__, __FILE__, __FUNCTION__, level, __VA_ARGS__);
+    log_impl(__LINE__, __FILE__, __func__, level, __VA_ARGS__);
 
 void log_impl(int line, const char *file, char const *function,
               log_level level, ...);

